src/Entity/Notice.php

```diff
@@ -4,42 +4,109 @@
 
 namespace App\Entity;
 
+use ApiPlatform\Core\Annotation\ApiProperty;
+use ApiPlatform\Core\Annotation\ApiResource;
 use App\EntityListener\NoticeListener;
 use App\Helper\NoticeVisibility;
 use App\Helper\StringHelper;
+use App\Serializer\V3\NormalizerOptions;
+use Closure;
 use DateTime;
 use DateTimeImmutable;
 use DateTimeInterface;
 use Doctrine\Common\Collections\ArrayCollection;
 use Doctrine\Common\Collections\Collection;
 use Doctrine\ORM\Mapping as ORM;
 use Symfony\Component\HttpFoundation\File\File;
+use Symfony\Component\Serializer\Annotation\Groups;
 use Symfony\Component\Validator\Constraints as Assert;
 use Vich\UploaderBundle\Mapping\Annotation as Vich;
 
 /**
- * Notice.
+ * A Notice holds a message written/contributed by a Contributor about a web page
+ * or a set of web pages selected by the matching contexts.
+ * This is the main content of the application.
  *
  * @ORM\Table(name="notice")
  * @ORM\Entity(repositoryClass="App\Repository\NoticeRepository")
  * @ORM\EntityListeners({NoticeListener::class})
  * @Vich\Uploadable
+ * @ApiResource(
+ *     normalizationContext={
+ *         "groups"={"read"},
+ *         NormalizerOptions::VERSION=4,
+ *     },
+ *     itemOperations={
+ *         "get"={
+ *             "normalization_context"={
+ *                 "groups"={"read"},
+ *                 NormalizerOptions::VERSION=4,
+ *             },
+ *         },
+ *         "delete"={
+ *             "access_control"="is_granted('can_delete', object)",
+ *         },
+ *     },
+ *     collectionOperations={
+ *         "get"={
+ *             "normalization_context"={
+ *                 "groups"={"read"},
+ *                 NormalizerOptions::VERSION=4,
+ *             },
+ *         },
+ *         "post"={
+ *             "denormalization_context"={
+ *                 "groups"={"create"},
+ *                 NormalizerOptions::VERSION=4,
+ *             },
+ *             "access_control"="is_granted('can_create', object)",
+ *         },
+ *     },
+ * )
  */
 class Notice
 {
+    // Goutte: Should we add <b><em><sup><sub> here as well?
+    public const ALLOWED_TAGS = '<p><a>';
+
     /**
-     * @var int
+     * A unique, incremental, numerical identifier for the Notice.
      *
+     * @var int
      * @ORM\Column(name="id", type="integer")
      * @ORM\Id
      * @ORM\GeneratedValue(strategy="AUTO")
+     * @Groups({"read"})
+     * @ApiProperty(
+     *     openapiContext={
+     *         "example"=42,
+     *     },
+     * )
      */
     private $id;
 
     /**
-     * @var ?string
+     * The visibility of the Notice.  See `NoticeVisibility`
+     * for an enumeration of the allowed values:
+     *   - "public": anyone may view this Notice
+     *   - "private": Notice is only visible to Contributor
+     *   - "archived": ???
+     *   - "draft": Notice is only visible to Contributor, pending publication
+     *   - "question": ???
```

Comment:
A question asked by a someone on a webpage. A question is not publicly visible.

Reply:
Thanks !